### PR TITLE
Cart functionality

### PR DESCRIPTION
**Pull request by Ankur9669**
In this i have added **add to cart, remove from cart, increment cart and decrement cart functionality**

Files to check
1. Cart.js
2. ProductLsiting.js
3. cart-reducer.js

### CODE REVIEW

File: fitbit-react/src/Context/products-context.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { createContext, useReducer, useContext } from "react";
+
+const ProductContext = createContext();
+const initialState = {};
+const reducer = (state, action) => {
```
Comment (bhtibrewal) [inline fitbit-react/src/Context/products-context.js]: Suggestion: Destructure action here itself

File: fitbit-react/src/Pages/ProductsPage/ProductListing/ProductListing.js [hunk 2]
```diff
@@ -21,7 +23,7 @@ function ProductListing() {
     return updatedData;
   };
   const getFilteredData = (filters) => {
-    let updatedData = [...products];
+    let updatedData = products.products;
     let equipmentCategory = [];
     let clothesCategory = [];
```
Comment (sri-rishi) [inline fitbit-react/src/Pages/ProductsPage/ProductListing/ProductListing.js]: Suggestion: you can use `const` declaration here

**Review by sri-rishi — APPROVED** [review]
UI suggestion: If cart is empty show that cart is empty and link to store or homepage like this

image

**Review by bhtibrewal — APPROVED** [review]
Everything looks okay to me, good to merge

**Comment by bhtibrewal** [comment]
> UI suggestion: If cart is empty show that cart is empty and link to store or homepage like this
> 
> image

Nice suggestion rishi

**Pull request merged**